commit cf5435dfb44f6f5961cf929947000feb82e4f132
Author: BinhNV <binhnv@exoplatform.com>
Date:   Mon Jul 30 18:09:50 2012 +0700

    commit code with custom view

diff --git a/bin/BookstoreMB.apk b/bin/BookstoreMB.apk
index e0d62a9..621927c 100644
Binary files a/bin/BookstoreMB.apk and b/bin/BookstoreMB.apk differ
diff --git a/bin/classes.dex b/bin/classes.dex
index f9980e1..796f28e 100644
Binary files a/bin/classes.dex and b/bin/classes.dex differ
diff --git a/bin/classes/com/example/bookstoremb/MainActivity.class b/bin/classes/com/example/bookstoremb/MainActivity.class
index 0dfe949..c017169 100644
Binary files a/bin/classes/com/example/bookstoremb/MainActivity.class and b/bin/classes/com/example/bookstoremb/MainActivity.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$drawable.class b/bin/classes/com/example/bookstoremb/R$drawable.class
index 2c2619d..5a6403a 100644
Binary files a/bin/classes/com/example/bookstoremb/R$drawable.class and b/bin/classes/com/example/bookstoremb/R$drawable.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$id.class b/bin/classes/com/example/bookstoremb/R$id.class
index 23690d7..dcffa59 100644
Binary files a/bin/classes/com/example/bookstoremb/R$id.class and b/bin/classes/com/example/bookstoremb/R$id.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$layout.class b/bin/classes/com/example/bookstoremb/R$layout.class
index 3c2cddc..68ef90d 100644
Binary files a/bin/classes/com/example/bookstoremb/R$layout.class and b/bin/classes/com/example/bookstoremb/R$layout.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$menu.class b/bin/classes/com/example/bookstoremb/R$menu.class
index 7c5bf8e..f2a4106 100644
Binary files a/bin/classes/com/example/bookstoremb/R$menu.class and b/bin/classes/com/example/bookstoremb/R$menu.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$string.class b/bin/classes/com/example/bookstoremb/R$string.class
index 259fdb8..ffe3f22 100644
Binary files a/bin/classes/com/example/bookstoremb/R$string.class and b/bin/classes/com/example/bookstoremb/R$string.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$style.class b/bin/classes/com/example/bookstoremb/R$style.class
index d203897..33838c3 100644
Binary files a/bin/classes/com/example/bookstoremb/R$style.class and b/bin/classes/com/example/bookstoremb/R$style.class differ
diff --git a/bin/classes/com/example/bookstoremb/adapter/BookstoreAdapter$ViewHolder.class b/bin/classes/com/example/bookstoremb/adapter/BookstoreAdapter$ViewHolder.class
new file mode 100644
index 0000000..b4d8e9b
Binary files /dev/null and b/bin/classes/com/example/bookstoremb/adapter/BookstoreAdapter$ViewHolder.class differ
diff --git a/bin/classes/com/example/bookstoremb/adapter/BookstoreAdapter.class b/bin/classes/com/example/bookstoremb/adapter/BookstoreAdapter.class
new file mode 100644
index 0000000..568b3f6
Binary files /dev/null and b/bin/classes/com/example/bookstoremb/adapter/BookstoreAdapter.class differ
diff --git a/bin/classes/com/example/bookstoremb/utils/RestClient$RequestMethod.class b/bin/classes/com/example/bookstoremb/utils/RestClient$RequestMethod.class
index 041691f..98055b9 100644
Binary files a/bin/classes/com/example/bookstoremb/utils/RestClient$RequestMethod.class and b/bin/classes/com/example/bookstoremb/utils/RestClient$RequestMethod.class differ
diff --git a/bin/classes/com/example/bookstoremb/utils/RestClient.class b/bin/classes/com/example/bookstoremb/utils/RestClient.class
index 081ee42..a6f847b 100644
Binary files a/bin/classes/com/example/bookstoremb/utils/RestClient.class and b/bin/classes/com/example/bookstoremb/utils/RestClient.class differ
diff --git a/bin/res/drawable-hdpi/cellbottom.9.png b/bin/res/drawable-hdpi/cellbottom.9.png
new file mode 100644
index 0000000..3889f43
Binary files /dev/null and b/bin/res/drawable-hdpi/cellbottom.9.png differ
diff --git a/bin/res/drawable-hdpi/cellsingle.9.png b/bin/res/drawable-hdpi/cellsingle.9.png
new file mode 100644
index 0000000..73aaa2f
Binary files /dev/null and b/bin/res/drawable-hdpi/cellsingle.9.png differ
diff --git a/bin/res/drawable-hdpi/celltop.9.png b/bin/res/drawable-hdpi/celltop.9.png
new file mode 100644
index 0000000..c58d919
Binary files /dev/null and b/bin/res/drawable-hdpi/celltop.9.png differ
diff --git a/bin/resources.ap_ b/bin/resources.ap_
index 9ae1770..bdad976 100644
Binary files a/bin/resources.ap_ and b/bin/resources.ap_ differ
diff --git a/gen/com/example/bookstoremb/R.java b/gen/com/example/bookstoremb/R.java
index f743182..f675e76 100644
--- a/gen/com/example/bookstoremb/R.java
+++ b/gen/com/example/bookstoremb/R.java
@@ -22,15 +22,20 @@ public final class R {
     public static final class drawable {
         public static final int airport=0x7f020000;
         public static final int arrow=0x7f020001;
-        public static final int ic_action_search=0x7f020002;
-        public static final int ic_launcher=0x7f020003;
+        public static final int cellbottom=0x7f020002;
+        public static final int cellsingle=0x7f020003;
+        public static final int celltop=0x7f020004;
+        public static final int customshape=0x7f020005;
+        public static final int ic_action_search=0x7f020006;
+        public static final int ic_launcher=0x7f020007;
     }
     public static final class id {
-        public static final int arrow=0x7f090005;
+        public static final int arrow=0x7f090006;
         public static final int content=0x7f090001;
-        public static final int icon=0x7f090003;
-        public static final int label=0x7f090004;
-        public static final int menu_settings=0x7f090006;
+        public static final int icon=0x7f090004;
+        public static final int label=0x7f090005;
+        public static final int list_row=0x7f090003;
+        public static final int menu_settings=0x7f090007;
         public static final int name=0x7f090000;
         public static final int nocontent=0x7f090002;
     }
diff --git a/res/drawable-hdpi/cellbottom.9.png b/res/drawable-hdpi/cellbottom.9.png
new file mode 100644
index 0000000..3d02511
Binary files /dev/null and b/res/drawable-hdpi/cellbottom.9.png differ
diff --git a/res/drawable-hdpi/cellsingle.9.png b/res/drawable-hdpi/cellsingle.9.png
new file mode 100644
index 0000000..4bf8432
Binary files /dev/null and b/res/drawable-hdpi/cellsingle.9.png differ
diff --git a/res/drawable-hdpi/celltop.9.png b/res/drawable-hdpi/celltop.9.png
new file mode 100644
index 0000000..c404e9f
Binary files /dev/null and b/res/drawable-hdpi/celltop.9.png differ
diff --git a/res/drawable/customshape.xml b/res/drawable/customshape.xml
new file mode 100644
index 0000000..ea5608a
--- /dev/null
+++ b/res/drawable/customshape.xml
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="rectangle">
+    
+	<corners android:radius="12px"/>
+	<gradient android:angle="90" android:endColor="#657383" android:startColor="#616D7E" android:type="linear"/>
+	
+</shape>
\ No newline at end of file
diff --git a/res/layout/activity_main.xml b/res/layout/activity_main.xml
index e0def63..db195dd 100644
--- a/res/layout/activity_main.xml
+++ b/res/layout/activity_main.xml
@@ -10,14 +10,14 @@
         android:layout_height="wrap_content"/>
     
     <ListView android:id="@android:id/list"
-        android:layout_width="fill_parent"
-        android:layout_height="fill_parent"
-        android:drawSelectorOnTop="false"
-        android:background="@color/white"
-        android:layout_marginLeft="20px"
-        android:layout_marginTop="20px"
-        android:layout_marginRight="20px"
-        android:layout_marginBottom="20px">
-    </ListView>
-
+		android:layout_width="fill_parent"
+	    android:layout_height="fill_parent"
+		android:drawSelectorOnTop="false"
+		android:background="@color/white"
+		android:layout_marginLeft="20px"
+		android:layout_marginTop="20px"
+		android:layout_marginRight="20px"
+		android:layout_marginBottom="20px">
+	</ListView>
+        
 </LinearLayout>
diff --git a/res/layout/row.xml b/res/layout/row.xml
index f0d0ed7..e627bc6 100644
--- a/res/layout/row.xml
+++ b/res/layout/row.xml
@@ -1,29 +1,31 @@
 <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="fill_parent"
-    android:layout_height="fill_parent" >
- 
-    <ImageView android:id="@+id/icon"
-        android:layout_width="wrap_content"
-        android:layout_height="fill_parent"
-        android:paddingLeft="2px"
-        android:paddingRight="2px"
-        android:paddingTop="8px"
-		android:src="@drawable/airport">
-	</ImageView>
-	
-	<TextView android:id="@+id/label"
-	    android:layout_width="wrap_content"
-	    android:layout_height="fill_parent"
-	    android:layout_toRightOf="@id/icon"
-		android:textSize="20sp" />
-	    
-	<ImageView android:id="@+id/arrow"
-	    android:layout_width="wrap_content"
-	    android:layout_height="fill_parent"
-	    android:paddingRight="2px"
-	    android:paddingTop="10px"
-		android:src="@drawable/arrow"
-		android:layout_alignParentRight="true">
-	</ImageView>
-	    
+    android:layout_height="fill_parent"
+    android:background="@drawable/cellsingle"
+    android:id="@+id/list_row">
+    	
+    	<ImageView android:id="@+id/icon"
+	        android:layout_width="wrap_content"
+	        android:layout_height="fill_parent"
+	        android:paddingLeft="2px"
+	        android:paddingRight="2px"
+	        android:paddingTop="8px"
+			android:src="@drawable/airport">
+		</ImageView>
+		
+		<TextView android:id="@+id/label"
+		    android:layout_width="wrap_content"
+		    android:layout_height="fill_parent"
+		    android:layout_toRightOf="@id/icon"
+			android:textSize="20sp" />
+		    
+		<ImageView android:id="@+id/arrow"
+		    android:layout_width="wrap_content"
+		    android:layout_height="fill_parent"
+		    android:paddingRight="2px"
+		    android:paddingTop="10px"
+			android:src="@drawable/arrow"
+			android:layout_alignParentRight="true">
+		</ImageView>
+		
 </RelativeLayout>
\ No newline at end of file
diff --git a/src/com/example/bookstoremb/MainActivity.java b/src/com/example/bookstoremb/MainActivity.java
index 02f7493..704087f 100644
--- a/src/com/example/bookstoremb/MainActivity.java
+++ b/src/com/example/bookstoremb/MainActivity.java
@@ -8,6 +8,7 @@ import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import android.R.layout;
 import android.app.ListActivity;
 import android.content.Intent;
 import android.os.Bundle;
@@ -17,8 +18,8 @@ import android.view.View;
 import android.widget.ArrayAdapter;
 import android.widget.ListView;
 import android.widget.TextView;
-import android.widget.Toast;
 
+import com.example.bookstoremb.adapter.BookstoreAdapter;
 import com.example.bookstoremb.models.Book;
 import com.example.bookstoremb.utils.Constants;
 import com.example.bookstoremb.utils.RestClient;
@@ -36,6 +37,8 @@ public class MainActivity extends ListActivity {
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        BookstoreAdapter adapter;
+        List<String> bookNames = new ArrayList<String>();
         books = new ArrayList<Book>();
         TextView nocontent = (TextView) findViewById(R.id.nocontent);
         try {
@@ -48,15 +51,15 @@ public class MainActivity extends ListActivity {
               Book book = Utils.createBookFromJSON(json);
               books.add(book);
             }
-            List<String> bookNames = new ArrayList<String>();
             for (Book b : books) {
               bookNames.add(b.getName());
             }
-            items = bookNames.toArray(new String[bookNames.size()]);
-            setListAdapter(new ArrayAdapter<String>(this, R.layout.row, R.id.label, items));
+            adapter = new BookstoreAdapter(this, bookNames);
+            this.setListAdapter(adapter);
             nocontent.setText("");
           } else {
-            setListAdapter(new ArrayAdapter<String>(this, R.layout.row, R.id.label, items));
+            adapter = new BookstoreAdapter(this, bookNames);
+            this.setListAdapter(adapter);
             nocontent.setText(R.string.nocontent);
           }
         } catch (JSONException jse) {
diff --git a/src/com/example/bookstoremb/adapter/BookstoreAdapter.java b/src/com/example/bookstoremb/adapter/BookstoreAdapter.java
new file mode 100644
index 0000000..180ea66
--- /dev/null
+++ b/src/com/example/bookstoremb/adapter/BookstoreAdapter.java
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2003-2012 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.example.bookstoremb.adapter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import android.graphics.Color;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.example.bookstoremb.MainActivity;
+import com.example.bookstoremb.R;
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : BinhNV
+ *          binhnv@exoplatform.com
+ * Jul 30, 2012  
+ */
+public class BookstoreAdapter extends BaseAdapter {
+
+  private MainActivity activity;
+  private List<String> arr = new ArrayList<String>();
+  private static LayoutInflater inflater;
+  
+  /**
+   * 
+   */
+  public BookstoreAdapter(MainActivity activity, List<String> arr) {
+    this.activity = activity;
+    this.arr = arr;
+    inflater = LayoutInflater.from(activity);
+  }
+
+  /* (non-Javadoc)
+   * @see android.widget.Adapter#getCount()
+   */
+  @Override
+  public int getCount() {
+    return arr.size();
+  }
+
+  /* (non-Javadoc)
+   * @see android.widget.Adapter#getItem(int)
+   */
+  @Override
+  public Object getItem(int position) {
+    return position;
+  }
+
+  /* (non-Javadoc)
+   * @see android.widget.Adapter#getItemId(int)
+   */
+  @Override
+  public long getItemId(int position) {
+    return position;
+  }
+
+  /* (non-Javadoc)
+   * @see android.widget.Adapter#getView(int, android.view.View, android.view.ViewGroup)
+   */
+  @Override
+  public View getView(int position, View convertView, ViewGroup parent) {
+    ViewHolder holder = null;
+    if (convertView == null) {
+      convertView = inflater.inflate(R.layout.row, null);
+      holder = new ViewHolder();
+      convertView.setTag(holder);
+    } else {
+      holder = (ViewHolder) convertView.getTag();
+    }
+    holder.textView = (TextView) convertView.findViewById(R.id.label);
+    holder.textView.setText(arr.get(position));
+    RelativeLayout listRow = (RelativeLayout) convertView.findViewById(R.id.list_row); 
+    if (position == 0) {
+      listRow.setBackgroundResource(R.drawable.celltop);
+//      listRow.setBackgroundColor(Color.RED);
+    }
+    if (position == (arr.size() - 1)) {
+      listRow.setBackgroundResource(R.drawable.cellbottom);
+//      listRow.setBackgroundColor(Color.GREEN);
+    }
+    return convertView;
+  }
+  
+  public static class ViewHolder {
+    public TextView textView;
+  }
+
+}
diff --git a/src/com/example/bookstoremb/utils/RestClient.java b/src/com/example/bookstoremb/utils/RestClient.java
index 229a795..638dbe3 100644
--- a/src/com/example/bookstoremb/utils/RestClient.java
+++ b/src/com/example/bookstoremb/utils/RestClient.java
@@ -31,8 +31,10 @@ import org.apache.http.NameValuePair;
 import org.apache.http.client.ClientProtocolException;
 import org.apache.http.client.HttpClient;
 import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpDelete;
 import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
 import org.apache.http.client.methods.HttpUriRequest;
 import org.apache.http.impl.client.DefaultHttpClient;
 import org.apache.http.message.BasicNameValuePair;
@@ -52,6 +54,7 @@ public class RestClient {
     GET, POST, PUT, DELETE;
   }
   
+  private final String AUTHORIZATION = "Authorization";
   private List<NameValuePair> params;
   private List<NameValuePair> headers;
   private String url;
@@ -88,7 +91,7 @@ public class RestClient {
             }
           }
           HttpGet request = new HttpGet(url + combinedParams);
-          request.setHeader("Authorization", getAuthorization());
+          request.setHeader(AUTHORIZATION, getAuthorization());
           for (NameValuePair h : headers) {
             request.addHeader(h.getName(), h.getValue());
           }
@@ -100,7 +103,7 @@ public class RestClient {
           HttpPost request = new HttpPost(url);
           for (NameValuePair h : headers) {
             request.addHeader(h.getName(), h.getValue());
-            request.setHeader("Authorization", getAuthorization());
+            request.setHeader(AUTHORIZATION, getAuthorization());
           }
           if (!params.isEmpty()) {
             request.setEntity(new UrlEncodedFormEntity(params, HTTP.UTF_8));
@@ -108,6 +111,26 @@ public class RestClient {
           executeRequest(request, url);
           break;
         }
+      case DELETE:
+        {
+          HttpDelete request = new HttpDelete(url);
+          request.setHeader(AUTHORIZATION, getAuthorization());
+          for (NameValuePair h : headers) {
+            request.addHeader(h.getName(), h.getValue());
+          }
+          executeRequest(request, url);
+          break;
+        }
+      case PUT:
+        {
+          HttpPut request = new HttpPut();
+          request.setHeader(AUTHORIZATION, getAuthorization());
+          for (NameValuePair h : headers) {
+            request.addHeader(h.getName(), h.getValue());
+          }
+          executeRequest(request, url);
+          break;
+        }
     }
   }
   
