commit 3dde7f3350cfac5a00edfef40f71509d803511dd
Author: BinhVN <binhnv@exoplatform.com>
Date:   Mon Jul 30 00:31:59 2012 +0700

    commit code with intent

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index b2f2232..84ad1ba 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -21,6 +21,7 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".ContentActivity"></activity>
     </application>
 
 </manifest>
\ No newline at end of file
diff --git a/bin/AndroidManifest.xml b/bin/AndroidManifest.xml
index b2f2232..84ad1ba 100644
--- a/bin/AndroidManifest.xml
+++ b/bin/AndroidManifest.xml
@@ -21,6 +21,7 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".ContentActivity"></activity>
     </application>
 
 </manifest>
\ No newline at end of file
diff --git a/bin/BookstoreMB.apk b/bin/BookstoreMB.apk
index 8d7df30..797675a 100644
Binary files a/bin/BookstoreMB.apk and b/bin/BookstoreMB.apk differ
diff --git a/bin/classes.dex b/bin/classes.dex
index ee8eb9a..ad45e8c 100644
Binary files a/bin/classes.dex and b/bin/classes.dex differ
diff --git a/bin/classes/com/example/bookstoremb/ContentActivity.class b/bin/classes/com/example/bookstoremb/ContentActivity.class
new file mode 100644
index 0000000..7663822
Binary files /dev/null and b/bin/classes/com/example/bookstoremb/ContentActivity.class differ
diff --git a/bin/classes/com/example/bookstoremb/MainActivity.class b/bin/classes/com/example/bookstoremb/MainActivity.class
index 9e95f99..90614f5 100644
Binary files a/bin/classes/com/example/bookstoremb/MainActivity.class and b/bin/classes/com/example/bookstoremb/MainActivity.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$id.class b/bin/classes/com/example/bookstoremb/R$id.class
index 5ed5fea..b0e00c6 100644
Binary files a/bin/classes/com/example/bookstoremb/R$id.class and b/bin/classes/com/example/bookstoremb/R$id.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$layout.class b/bin/classes/com/example/bookstoremb/R$layout.class
index c0f13d9..bc7379a 100644
Binary files a/bin/classes/com/example/bookstoremb/R$layout.class and b/bin/classes/com/example/bookstoremb/R$layout.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$menu.class b/bin/classes/com/example/bookstoremb/R$menu.class
index 8083d9f..356caad 100644
Binary files a/bin/classes/com/example/bookstoremb/R$menu.class and b/bin/classes/com/example/bookstoremb/R$menu.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$string.class b/bin/classes/com/example/bookstoremb/R$string.class
index fa40d77..fae81ac 100644
Binary files a/bin/classes/com/example/bookstoremb/R$string.class and b/bin/classes/com/example/bookstoremb/R$string.class differ
diff --git a/bin/classes/com/example/bookstoremb/R$style.class b/bin/classes/com/example/bookstoremb/R$style.class
index 5b23dac..88fff7c 100644
Binary files a/bin/classes/com/example/bookstoremb/R$style.class and b/bin/classes/com/example/bookstoremb/R$style.class differ
diff --git a/bin/jarlist.cache b/bin/jarlist.cache
new file mode 100644
index 0000000..1b5ec3f
--- /dev/null
+++ b/bin/jarlist.cache
@@ -0,0 +1,3 @@
+# cache for current jar dependecy. DO NOT EDIT.
+# format is <lastModified> <length> <SHA-1> <path>
+# Encoding is UTF-8
diff --git a/bin/resources.ap_ b/bin/resources.ap_
index 387f210..ce5e828 100644
Binary files a/bin/resources.ap_ and b/bin/resources.ap_ differ
diff --git a/gen/com/example/bookstoremb/R.java b/gen/com/example/bookstoremb/R.java
index 1e34e31..0561043 100644
--- a/gen/com/example/bookstoremb/R.java
+++ b/gen/com/example/bookstoremb/R.java
@@ -26,14 +26,16 @@ public final class R {
         public static final int ic_launcher=0x7f020003;
     }
     public static final class id {
-        public static final int arrow=0x7f090002;
-        public static final int icon=0x7f090000;
-        public static final int label=0x7f090001;
-        public static final int menu_settings=0x7f090003;
+        public static final int arrow=0x7f090003;
+        public static final int icon=0x7f090001;
+        public static final int label=0x7f090002;
+        public static final int menu_settings=0x7f090004;
+        public static final int name=0x7f090000;
     }
     public static final class layout {
-        public static final int activity_main=0x7f030000;
-        public static final int row=0x7f030001;
+        public static final int activity_content=0x7f030000;
+        public static final int activity_main=0x7f030001;
+        public static final int row=0x7f030002;
     }
     public static final class menu {
         public static final int activity_main=0x7f080000;
diff --git a/res/layout/activity_content.xml b/res/layout/activity_content.xml
new file mode 100644
index 0000000..9aa0de8
--- /dev/null
+++ b/res/layout/activity_content.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="fill_parent"
+    android:layout_height="fill_parent" 
+    android:orientation="vertical"
+    android:background="@color/grey" >
+    
+    <LinearLayout android:layout_width="fill_parent"
+        android:layout_height="fill_parent"
+        android:drawSelectorOnTop="false"
+        android:background="@color/white"
+        android:layout_marginLeft="20px"
+        android:layout_marginTop="20px"
+        android:layout_marginRight="20px"
+        android:layout_marginBottom="20px">
+        
+        <TextView android:id="@+id/name"
+	    android:layout_width="wrap_content"
+	    android:layout_height="wrap_content"/>
+        
+    </LinearLayout>
+    
+</LinearLayout>
\ No newline at end of file
diff --git a/src/com/example/bookstoremb/ContentActivity.java b/src/com/example/bookstoremb/ContentActivity.java
new file mode 100644
index 0000000..96f13c6
--- /dev/null
+++ b/src/com/example/bookstoremb/ContentActivity.java
@@ -0,0 +1,24 @@
+/**
+ * 
+ */
+package com.example.bookstoremb;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+/**
+ * @author Administrator
+ *
+ */
+public class ContentActivity extends Activity {
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_content);
+		TextView str = (TextView) findViewById(R.id.name);
+		str.setText("Hello world");
+	}
+
+}
diff --git a/src/com/example/bookstoremb/MainActivity.java b/src/com/example/bookstoremb/MainActivity.java
index 767a65e..d8fd426 100644
--- a/src/com/example/bookstoremb/MainActivity.java
+++ b/src/com/example/bookstoremb/MainActivity.java
@@ -15,6 +15,7 @@ import org.apache.http.impl.client.BasicResponseHandler;
 import org.apache.http.impl.client.DefaultHttpClient;
 
 import android.app.ListActivity;
+import android.content.Intent;
 import android.os.Bundle;
 import android.util.Base64;
 import android.view.Menu;
@@ -36,26 +37,26 @@ public class MainActivity extends ListActivity {
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        HttpClient client = new DefaultHttpClient();
-        HttpGet request = new HttpGet(URL);
-        request.setHeader("Authorization",
-                      "Basic "
-                          + Base64.encodeToString((username + ":" + password).getBytes(),
-                                                  Base64.NO_WRAP));
-        ResponseHandler<String> handler = new BasicResponseHandler();
-        try {
-          HttpResponse response = client.execute(request);
-          HttpEntity entity = response.getEntity();
-          if (entity != null) {
-            InputStream is = entity.getContent();
-            String responseStr = convertStreamToString(is);
-          }
-        } catch (ClientProtocolException cpe) {
-          cpe.printStackTrace();
-        } catch (IOException e) {
-          e.printStackTrace();
-        }
-        client.getConnectionManager().shutdown();
+//        HttpClient client = new DefaultHttpClient();
+//        HttpGet request = new HttpGet(URL);
+//        request.setHeader("Authorization",
+//                      "Basic "
+//                          + Base64.encodeToString((username + ":" + password).getBytes(),
+//                                                  Base64.NO_WRAP));
+//        ResponseHandler<String> handler = new BasicResponseHandler();
+//        try {
+//          HttpResponse response = client.execute(request);
+//          HttpEntity entity = response.getEntity();
+//          if (entity != null) {
+//            InputStream is = entity.getContent();
+//            String responseStr = convertStreamToString(is);
+//          }
+//        } catch (ClientProtocolException cpe) {
+//          cpe.printStackTrace();
+//        } catch (IOException e) {
+//          e.printStackTrace();
+//        }
+//        client.getConnectionManager().shutdown();
         setListAdapter(new ArrayAdapter<String>(this, R.layout.row, R.id.label, items));
     }
 
@@ -71,6 +72,8 @@ public class MainActivity extends ListActivity {
     @Override
     protected void onListItemClick(ListView l, View v, int position, long id) {
       super.onListItemClick(l, v, position, id);
+      Intent intent = new Intent(this, ContentActivity.class);
+      startActivity(intent);
     }
     
     private String convertStreamToString(InputStream is) {
